### ISSUE TITLE
Sense App Real-Time view sluggish while node API socket is running

### ISSUE THREAD

**Issue by brbeaird** [brbeaird/SmartThings_SenseMonitor#4]
I've discovered that if I have the node server running and pulling Sense data, the real-time monitor in the Sense mobile app (or web view) is basically unusable. If I kill the node server, the real-time monitor behaves like normal, showing instant updates.

This did not used to be the case. Previously, the real-time view was unaffected by other running API's. If anything, the mobile app connection trumped everything else as when I would close the mobile app, it would close any API connections that were open, forcing a reauth event.

I've submitted an e-mail to Sense support asking for their thoughts on this. Given the unofficial nature by which we're connecting, I'm not exactly sure what will happen here.

**Comment by brbeaird** [comment]
Finally figured out a fix for this. I've modified the sense library so the websocket connection can be directly opened/closed. With that in place, I've also altered the SmartThings process to open the websocket (re-using the token, not re-authenticating) once every 2 minutes and then immediately close it when we get a set of data. I'll only have it re-auth once the token is no longer valid.

This should drastically cut down on traffic flowing from Sense and ease the issues affecting the other real-time monitors.  I'm testing it now on my machine and will let it run for a day or so and verify everything is stable.
